packages/redshift/sodasql/dialects/redshift_dialect.py

```diff
@@ -26,7 +26,8 @@ class RedshiftDialect(PostgresDialect):
     def get_aws_credentials_optional(parser: Parser):
         access_key_id = parser.get_str_optional_env('access_key_id')
         role_arn = parser.get_str_optional_env('role_arn')
-        if access_key_id or role_arn:
+        profile_name = parser.get_str_optional_env('profile_name')
```

Comment:
This seems to be unused in credentials, I think you missed passing it to AWSCredentials